docs/github.md

```diff
@@ -0,0 +1,24 @@
+# Contributing to Calyx
+
+## A few notes on Github practices around the Calyx project
```

Comment:
"Github Workflow"

Also, couple of things that come to mind:
- Formatting (both Rust and Python)
- Dockerfile builds: If you change the dockerfile, it will cause the CI to automatically rebuild it and run your tests on it.